python/dgl/nn/pytorch/explain/pgexplainer.py

```diff
@@ -357,6 +423,182 @@ def explain_graph(self, graph, feat, tmp=1.0, training=False, **kwargs):
+            Classification probabilities given the masked graph. It is a tensor of
+            shape :math:`(B, L)`, where :math:`L` is the different types of label
+            in the dataset, and :math:`B` is the batch size.
+        Tensor
+            Edge weights which is a tensor of shape :math:`(E)`, where :math:`E`
+            is the number of edges in the graph. A higher weight suggests a larger
+            contribution of the edge.
+        DGLGraph
+            The batched set of subgraphs induced on the k-hop in-neighborhood
+            of the input center nodes.
+        Tensor
+            The batched list of node features.
+        Tensor
+            The new IDs of the subgraph center nodes.
+
+        Examples
+        --------
+
+        >>> import dgl
+        >>> import numpy as np
+        >>> import torch
+
+        >>> # Define the model
+        >>> class Model(torch.nn.Module):
+        ...     def __init__(self, in_feats, out_feats):
+        ...         super().__init__()
+        ...         self.conv1 = dgl.nn.GraphConv(in_feats, out_feats)
+        ...         self.conv2 = dgl.nn.GraphConv(out_feats, out_feats)
+        ...
+        ...     def forward(self, g, h, embed=False, edge_weight=None):
+        ...         h = self.conv1(g, h, edge_weight=edge_weight)
+        ...         if embed:
+        ...             return h
+        ...         return self.conv2(g, h)
+
+        >>> # Load dataset
+        >>> data = dgl.data.CoraGraphDataset()
+        >>> g = data[0]
+        >>> features = g.ndata["feat"]
+        >>> labels = g.ndata["label"]
+
+        >>> # Train the model
+        >>> model = Model(features.shape[1], data.num_classes)
+        >>> criterion = torch.nn.CrossEntropyLoss()
+        >>> optimizer = torch.optim.Adam(model.parameters(), lr=1e-2)
+        >>> for epoch in range(20):
+        ...     logits = model(g, features)
+        ...     loss = criterion(logits, labels)
+        ...     optimizer.zero_grad()
+        ...     loss.backward()
+        ...     optimizer.step()
+
+        >>> # Initialize the explainer
+        >>> explainer = dgl.nn.PGExplainer(
+        ...     model, data.num_classes, num_hops=2, explain_graph=False
+        ... )
+
+        >>> # Train the explainer
+        >>> # Define explainer temperature parameter
+        >>> init_tmp, final_tmp = 5.0, 1.0
+        >>> optimizer_exp = torch.optim.Adam(explainer.parameters(), lr=0.01)
+        >>> epochs = 10
+        >>> for epoch in range(epochs):
+        ...     tmp = float(init_tmp * np.power(final_tmp / init_tmp, epoch / epochs))
+        ...     for node in g.nodes():
+        ...         loss = explainer.train_step_node(node, g, features, tmp)
+        ...         optimizer_exp.zero_grad()
+        ...         loss.backward()
+        ...         optimizer_exp.step()
+
+        >>> # Explain the prediction for graph 0
+        >>> probs, edge_weight, bg, bf, inverse_indicies = explainer.explain_node(
```

Comment:
When do users need `bg, bf, inverse_indicies`?

Reply:
I followed return values similar to [`gnn_explainer`](https://github.com/dmlc/dgl/blob/master/python/dgl/nn/pytorch/explain/gnnexplainer.py#L317C9-L317C9), but after some review if it's safe to assume root of each subgraph is node `0` then this can be omitted and `bg` can be unbatched to retrieve every subgraph matching features

Reply:
mainly used internally for train step on batched graph*

```py
prob, _, batched_graph, inverse_indices = self.explain_node(
    nodes, graph, feat, tmp=tmp, training=True, **kwargs
)
pred = self.model(batched_graph, batched_feats, embed=False, **kwargs)
pred = pred.argmax(-1).data

loss = self.loss(prob[inverse_indices], pred[inverse_indices])
```

Reply:
If this is mainly for internal use, then maybe there is no need to return them.